Finish readme and add component diagram (#9)
Harmos274/gotrans: README.md

camion_b 3 4 4000 5 -- Truck name, X and Y position, max weight and cycle and cooldown after loading.
```

## Repository design

The sources are organised through 2 packages, the main package, `gotrans`, located at the root of the
directory, and its internal package `warehouse`.

The role of the `gotrans` package is to manage the communication with the user it contains the entry point
of the program in the `gotrans.go` file.

> There's also the `parse_input_file.go` file, whose role is to help the parsing of the input file given by

Reviewer: ```suggestion
There's also the `parse_input_file.go` file, whose role is to handle the parsing of the input file given by
```

the users inputs, the `show_warehouse.go` that contains everything needed to print the warehouse on the

> terminal and the `graphical.go` that contains the functions needed to run the graphical UI.

Reviewer: ```suggestion
terminal and `graphical.go` that contains the functions needed to run the graphical UI.
```

In the `warehouse` package, the `warehouse.go` file contains the description of the Warehouse and the
functions to modify its data. The `event.go` file describes all the events occurring during the warehouse
cleaning execution cycles. Finally, the `al.go` file contains the pathfinding algorithm used in the cleaning
warehouse process.

## Pathfinding strategy

> For each `forklift` the algorithm find the quickest path to go to every `package` in the `warehouse`,

Reviewer: ```suggestion
For each `forklift`, the algorithm finds the quickest path to go to every `package` in the `warehouse`,
```

> if the `forklift` find a shorter path than another, its path is chosen, otherwise the `package` is

Reviewer: ```suggestion
if the `forklift` finds a shorter path than another, its path is chosen, otherwise the `package` is
```

removed from the `forklift`'s targets and another `package` is selected.

> Once the `forklift` arrives by the `package` it picks it up and search the quickest path to the `truck`.

Reviewer: ```suggestion
Once the `forklift` arrives by the `package` it picks it up and searches the quickest path to the `truck`.
```

When the `forklift` arrives by the `truck` it loads its `package` in the `truck`, if possible, otherwise it waits.

> Once the `package` has been delivered the `forklift` go to another targets if there's one.

Reviewer: ```suggestion
Once the `package` has been delivered the `forklift` goes to another targets if there is one.
```
